### PR TITLE
fix: onetime walk retry

### PR DESCRIPTION
**Pull request by omrozowicz-splunk**
# Description

If snmp simulator is down, then onetime walk fails. We need data from onetime_walk for enricher to work. This PR adds retries for onetime_walk.

Fixes # 43365

## Type of change

Please delete options that are not relevant.

- [x] Bug fix

## How Has This Been Tested?

Disable snmp simulator and install SC4SNMP. Set onetime_task_frequency parameter to for ex. 3, so it runs more frequent. After some time when original onetime_walk fails, bring the simulator up. Check if in ~3 minutes another onetime walk starts.

## Checklist

- [x] My commit message is [conventional](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] I have run pre-commit on all files before creating the PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: splunk_connect_for_snmp_poller/mongo.py [hunk 1]
```diff
@@ -108,6 +111,26 @@ def contains_host(self, host):
...
+            {
+                "_id": host,
+                "host": host,
+                "version": version,
+                "community": community,
+            }
+        )
+
+    def delete_onetime_walk_result(self, host):
+        logger.info("delete_onetime_walk_result")
```
Comment (okuzhel) [inline splunk_connect_for_snmp_poller/mongo.py]: maybe make it into 1 log? :)

File: splunk_connect_for_snmp_poller/mongo.py [hunk 2]
```diff
@@ -108,6 +111,26 @@ def contains_host(self, host):
     def add_host(self, host):
         self._walked_hosts.insert_one({"_id": host})
 
+    def get_all_unwalked_hosts(self):
+        return self._unwalked_hosts.find({})
```
Comment (okuzhel) [inline splunk_connect_for_snmp_poller/mongo.py]: find return cursor, to close it we should pack into into the list()

**Review by okuzhel — APPROVED** [review]

**Review by weliasz — APPROVED** [review]

File: splunk_connect_for_snmp_poller/mongo.py (hunk 1 above)
Comment (omrozowicz-splunk) [reply splunk_connect_for_snmp_poller/mongo.py]: Updated, also for added host

File: splunk_connect_for_snmp_poller/mongo.py (hunk 2 above)
Comment (omrozowicz-splunk) [reply splunk_connect_for_snmp_poller/mongo.py]: Good point, fixed

**Pull request merged**